numpy/core/src/multiarray/ctors.c

```diff
@@ -2105,11 +2105,32 @@ PyArray_FromStructInterface(PyObject *input)
         }
     }
 
+    /* a tuple to hold references */
+    PyObject *refs = PyTuple_New(2);
+    if (!refs) {
+        PyErr_SetString(PyExc_MemoryError, "Failed to allocate a Tuple of size 2");
```

Comment:
The error should already be set, and since it is an allocation error the less done after that the better.
```suggestion
```

Reply:
Ah, OK. Thanks for pointing this out. I made the suggested change.